doc/features/pre-existing-tenant.md

```diff
@@ -0,0 +1,59 @@
+# Pre-existing Tenant Feature
+
+# Table of Contents
+
+- [1. Overview](#1-overview)
+- [2. Motivation](#2-motivation)
+- [3. Mechanism](#3-mechanism)
+- [4. Lifecycle](#4-lifecycle)
+- [5. Example](#5-example)
+
+## 1. Overview
+
+This feature is introduced to enable adding a Kubernetes cluster to a pre-existing tenant like the common tenant on APIC. The APIC objects provisioned for different clusters should be uniquely named and distinguishable. When deleting a cluster, the objects created by other clusters shouldn’t be affected.
+
+## 2. Motivation
+
+The pre-existing tenant feature be able to process the tenant field in ACI config of the input yaml file to provision the Kubernetes cluster under the given tenant. This includes creating create application profile, EPGs, BDs and required contracts under the existing tenant. If this tenant isn’t already created, our tool will create it, but not delete it in the un-provision step. All other APIC objects created for this cluster being un-provisioned will be deleted. The only case where the tool will delete a tenant is when the tenant was created by our tool (identified by a unique tag), and has no other clusters provisioned under it.
```

Comment:
By enabling the pre-existing tenant feature it is possible to provision the Kubernetes cluster under a given tenant. This overrides the default behaviour where a Tenant is created with the same name as the cluster. 

_The only case where the tool will delete a tenant is when the tenant was created by our tool (identified by a unique tag), and has no other clusters provisioned under it._ ==> Are we saying the the tenant is deleted only if the "default" tenant creation mode is created? 

- I do not think we need to state here that is enabled in the YAML file as anyway we show it below.